test/jest/cookieSyncManager.spec.ts

```diff
@@ -451,4 +493,43 @@ describe('CookieSyncManager', () => {
             expect(cookieSyncDates[42]).toBeUndefined();
         });
     });
+
+    describe('#combineUrlWithRedirect', () => {
+        it('should combine a pixelUrl with a redirectUrl', () => {
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [pixelSettings],
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+
+            const result = cookieSyncManager.combineUrlWithRedirect(
+                '1234',
+                'https://test.com',
+                '?redirect=https://redirect.com&mpid=%%mpid%%',
+            );
+
+            expect(result).toBe('https://test.com%3Fredirect%3Dhttps%3A%2F%2Fredirect.com%26mpid%3D1234');
```

Comment:
something looks off to me here.  perhaps it's the example you are using for the redirect url that is causing this.  but the urlWithRedirect is what the browser will load, and in this case, I don't think the browser will use the %3F and know that query parameters are coming after it. maybe it's best to look at some [real examples](https://jssdkcdns.mparticle.com/JS/v2/beab3f4d34281d45bfcdbbd7eb21c083/config?env=0)  (search for `pixelConfig`)and see if you need to make any changes

Reply:
`%3F` is what will be returned when you run the `redirect` url through `encodeURIComponent`.

Reply:
Not sure if this is a bug or something we didn't consider, but the current logic does not really append the redirect url as a query param. It just concatenates `url` and `redirectUrl` into a single string. From the examples provided, it looks like it's up to the customer/user to add a `?` into the url before concatenation. Is this expected?